Automattic/simplenote-ios
# Drops SAMKeychain: Mark 1

### Details:
- New **SimplenoteKeychain** namespace, which is meant to hold all the keychain constants
- Switches **SAMKeychain** >> **KeychainManager** for PIN Lock code access operations

@aerych Can I bug you with this one sir?
Thank you!!!

P.s.: I've been going back and forth with the idea of adding log statements. WDYT? 
Perhaps it's best to patch the Keychain Wrapper to log errors, rather than adding `do / catch` all over?

Ref. #880

---

### Test: Pin Lock!
1. Install **develop** and log into your account
2. Open **Settings** > **Passcode**
3. Set a Pin Lock code
4. Set the `Lock Timeout` to 15 seconds
5. Install this branch on top

- [x] Verify the Pin Lock code you've used in (3) unlocks the app
- [x] Verify that the Pin Lock is requested only if the app stays for +15 seconds backgrounded 
- [x] Verify that the app remains unlocked if you reopen in under 15 seconds

### Release
These changes do not require release notes.
